### DISCUSSION TITLE
Run X11 apps in windows like crostini?

### DISCUSSION THREAD

**Discussion by aachrisg** [dnschneid/crouton#4650] [General]
Hi, just discovered this package and trying it out on my pixel slate with debian bullseye. I didn't have any real problems getting it to work, and compiling all my c++ code on it.

For running X11 GUI apps, is there any way to get them to run seamlessly like they do when run under crostini? In crostini, if I run emacs, it pops up in its own window that is nicely integrated with the normal ChromeOS windows, and if I run another app, up pops another window.

**Comment by TheTechRobo** [comment]
It's not seamless but from inside the chroot you can run `xiwi <COMMAND>` so for emacs, `xiwi emacs`

It's not very fast tho

**Reply by aachrisg** [reply]
Yes, I tried that. It works but like you say it's slow, and also the apps don't fit into the chromeos window system nearly as well as under crostini.
I wonder what crostini is doing - I'd assumed it was just talking to some built in X server in ChromeOS, but doing export DISPLAY=127.0.0.1:0 from a crouton shell didn't work - no server to connect to.

**Reply by aachrisg** [reply]
..Maybe if I run crostini _also_, apps running on crouton could talk to its X server..hmmm..

**Reply by TheTechRobo** [reply]
crostini uses somellier, there's a pull request to add support (#4410) but I haven't tested it.
